src/app_engine/probers.py

```diff
@@ -94,18 +99,74 @@ def get(self):
     self.handle_ceod_response(error_message, status_code)
 
 class ProbeColliderPage(webapp2.RequestHandler):
-  def handle_collider_response(self, error_message, status_code):
-    self.response.set_status(status_code)
+  def handle_collider_response(
+      self, error_message, status_code, collider_instance):
+    result = {
+        constants.WSS_HOST_INSTANCE_KEY: collider_instance,
+        constants.WSS_HOST_STATUS_CODE_KEY: status_code
+    }
     if error_message is not None:
-      send_alert_email('Collider Error', error_message)
+      send_alert_email('Collider Error', error_message, collider_instance)
 
       logging.warning('Collider prober error: ' + error_message)
-      self.response.out.write(error_message)
+      result[constants.WSS_HOST_ERROR_MESSAGE_KEY] = error_message
+      result[constants.WSS_HOST_STATUS_KEY] = False
     else:
-      self.response.out.write('Success!')
+      result[constants.WSS_HOST_STATUS_KEY] = True
+    return result
+
+  def store_instance_state(self, results):
+    # Store a list with status for each collider host.
+    # The list should be in order of decreasing uptime (the first server in
+    # the list has been up the longest).
+    memcache_client = memcache.Client()
+    for retries in xrange(constants.MEMCACHE_RETRY_LIMIT):
+      collider_status = memcache_client.gets(constants.WSS_HOST_STATUS_LIST_KEY)
+      if collider_status is None:
+        memcache_client.set(constants.WSS_HOST_STATUS_LIST_KEY, [])
+        collider_status = memcache_client.gets(
+            constants.WSS_HOST_STATUS_LIST_KEY)
+      collider_status = self.create_collider_status_list(
+          collider_status, results)
+      if memcache_client.cas(constants.WSS_HOST_STATUS_LIST_KEY,
+                             collider_status):
+        logging.info('collider status saved to memcache')
+        break
+      logging.warning('retry # ' + str(retries) + ' to set collider status')
+
+  def create_collider_status_list(self, existing_record, results):
+    # Keep only the instance name and status.
+    sorted_results = [{
```

Comment:
If you store tuples instead you can avoid all this sorting logic. You can use WSS_HOST_INSTANCE_INDEX, WSS_HOST_STATUS_INDEX instead.